### PR TITLE
image: rework dockerclient to not depend on internal types

### PR DESCRIPTION
**Pull request by mfojtik**
This allows us to move it to staging and `oc new-app` and `oc new-build` can start using today.

### CODE REVIEW

File: pkg/image/importer/dockerv1client/scheme.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// this is the only entrypoint which deals in github.com/fsouza/go-dockerclient.Image and expects to use our conversion capability to coerce an external
+// type into an api type.  Localize the crazy here.
+// TODO this scheme needs to be configurable or we're going to end up with weird problems.
+func init() {
+	utilruntime.Must(imagev1.Install(DockerClientScheme))
+	utilruntime.Must(imageapi.Install(DockerClientScheme))
+
+	err := DockerClientScheme.AddConversionFuncs(
+		// Convert docker client object to internal object
+		func(in *docker.Image, out *imageapi.DockerImage, s conversion.Scope) error {
```
Comment (deads2k) [inline pkg/image/importer/dockerv1client/scheme.go]: this is a move?

File: pkg/image/importer/dockerv1client/scheme.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+// this is the only entrypoint which deals in github.com/fsouza/go-dockerclient.Image and expects to use our conversion capability to coerce an external
+// type into an api type.  Localize the crazy here.
+// TODO this scheme needs to be configurable or we're going to end up with weird problems.
+func init() {
+	utilruntime.Must(imagev1.Install(DockerClientScheme))
+	utilruntime.Must(imageapi.Install(DockerClientScheme))
+
+	err := DockerClientScheme.AddConversionFuncs(
```
Comment (deads2k) [inline pkg/image/importer/dockerv1client/scheme.go]: nit: utilruntime.Must

File: staging/src/github.com/openshift/oc/pkg/helpers/dockerclient/client.go [hunk 3]
```diff
@@ -0,0 +1,955 @@
+package dockerv1client
```
Comment (deads2k) [inline staging/src/github.com/openshift/oc/pkg/helpers/dockerclient/client.go]: this is a copy?  it all looks external, why not a candidate for library-go?

Comment (mfojtik) [reply staging/src/github.com/openshift/oc/pkg/helpers/dockerclient/client.go]: it is mostly copy bu without the docker types and crazy conversions... the auth and interface is all the same. i will try to make the changes more obvious in commit

File: pkg/image/importer/dockerv1client/scheme.go (hunk 1 above)
Comment (mfojtik) [reply pkg/image/importer/dockerv1client/scheme.go]: schema was extracted from the client.go to make it more visible and reworked to not poison the legacyscheme.

**Comment by mfojtik** [comment]
/hold

This is a dead end. The image apis need complete rework to get rid of the mess we introduced by mixing openshift apis and fsouza api. It needs clear separation and we need to get rid of conversions and other black magic.

@deads2k this will probably keep me busy till end of this week....

**Pull request closed without merging**